.github/workflows/integration_tests_app_ci.yml

```diff
@@ -101,85 +97,63 @@ jobs:
+          # encode them in the workflow file, because we could easily leak the
+          # base64 encoded version in logs.
+          USER_1_EMAIL_BASE64: ${{ secrets.INTEGRATION_TEST_USER_1_EMAIL_DART_DEFINE_BASE64 }}
+          USER_1_PASSWORD_BASE64: ${{ secrets.INTEGRATION_TEST_USER_1_PASSWORD_DART_DEFINE_BASE64 }}
         run: |
-          API_LEVEL=33
-          TARGET=google_apis
-          ARCH=x86_64
-
-          echo "Installing system image"
-          echo "y" | $ANDROID_SDK_ROOT/cmdline-tools/latest/bin/sdkmanager "system-images;android-$API_LEVEL;$TARGET;$ARCH"
-
-          echo "Creating AVD"
-          echo "no" | $ANDROID_SDK_ROOT/cmdline-tools/latest/bin/avdmanager create avd -n test_emulator -k "system-images;android-$API_LEVEL;$TARGET;$ARCH" --force
+          # Flutter build is required to generate files in android/ to build the
+          # gradle project.
+          #
+          # We are using the prod flavor because we were not able to set up
+          # Firebase Test Lab with the dev flavor. We always got "No tests
+          # found.".
+          flutter build apk \
+            --target=lib/main_prod.dart \
+            --flavor prod
 
-          echo "Starting emulator"
-          $ANDROID_SDK_ROOT/emulator/emulator -avd test_emulator -no-snapshot -no-boot-anim -no-window -camera-back virtualscene -camera-front emulated &
-          adb wait-for-device shell 'while [[ -z $(getprop sys.boot_completed) ]]; do sleep 1; done; input keyevent 82'
+          ./gradlew app:assembleProdDebugAndroidTest
+          ./gradlew app:assembleProdDebug \
+            -Ptarget=integration_test/app_test.dart \
+            -Pdart-defines="$USER_1_EMAIL_BASE64,$USER_1_PASSWORD_BASE64"
 
-      - name: Run integration tests
+      - name: Setup credentials
         env:
-          USER_1_EMAIL: ${{ secrets.INTEGRATION_TEST_USER_1_EMAIL }}
-          USER_1_PASSWORD: ${{ secrets.INTEGRATION_TEST_USER_1_PASSWORD }}
+          TEST_LAB_CREDENTIALS: ${{ secrets.FIREBASE_TEST_LAB_DEV_KEY }}
         run: |
-          # We need to run the integration tests with the prod flavor because
-          # using not the default flavor will cause an exception when
-          # uninstalling the app, see:
-          # https://github.com/flutter/flutter/issues/88690
-          flutter test \
-            integration_test \
-            --flavor prod \
-            --dart-define USER_1_EMAIL=$USER_1_EMAIL \
-            --dart-define USER_1_PASSWORD=$USER_1_PASSWORD
+          echo $TEST_LAB_CREDENTIALS > firebase-test-lab.json
+          gcloud auth activate-service-account --key-file=firebase-test-lab.json
+          gcloud --quiet config set project sharezone-debug
+
+      # Runs our integration tests with Firebase Test Lab.
+      #
+      # Advantages of Firebase Test Lab:
+      #  * We can use a Linux runner instead of a macOS runner (macOS runner is
+      #    for an Android emulator required).
```

Comment:
```suggestion
      #  * We can use a Linux runner instead of a macOS runner in 
      #    Github Actions (macOS runner is required for using an Android emulator).
```